admin/src/app/Components/registration/registration.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component } from '@angular/core';
+import { MatDialogRef } from '@angular/material/dialog';
+import { AbstractControl, FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+
+import { AuthService } from 'src/app/auth.service';
+
+@Component({
+  selector: 'app-registration',
+  templateUrl: './registration.component.html',
+  styleUrls: ['./registration.component.css']
+})
+export class RegistrationComponent{
+  public registrationForm: FormGroup;
+  public unsuccessWarning: boolean = false;
+
+  public constructor(
+    private readonly authService: AuthService,
+    private readonly matDialogRef: MatDialogRef<RegistrationComponent>,
+    private readonly formBuilder: FormBuilder,
+    private readonly snackBar: MatSnackBar) {
+    this.registrationForm = this.formBuilder.group({
+      email: [
+        '',
+        [
+          Validators.required,
+          Validators.email
+        ]
+      ],
+      password: ['', [Validators.required]]
+    });
+  }
+
+  public get email(): AbstractControl | null {
+    return this.registrationForm.get('email');
+  }
+
+  public get password(): AbstractControl | null {
+    return this.registrationForm.get('password');
+  }
+
+  public closeRegistrationDialog(): void {
+    this.matDialogRef.close();
+  }
+
+  public openSuccessSnackBar(message: string): void {
+    this.snackBar.open(
+      `${message}`,
+      'Confirm'
+    );
+  }
+
+  public register(email: string, password: string): void {
+    this.authService.register(email, password)
+    .subscribe(
+      () => {
+        this.unsuccessWarning = false;
+        this.openSuccessSnackBar('Successfully created!');
+      },
+      () => {
+        this.unsuccessWarning = true;
+      }
+    );
```

Comment:
I think you decided to use different code style approach for chained methods.
But even with the current choice it is not correct.

Current style
```suggestion
    this.authService.register(email, password)
      .subscribe(
        () => {
          this.unsuccessWarning = false;
          this.openSuccessSnackBar('Successfully created!');
        },
        () => {
          this.unsuccessWarning = true;
        }
      );
```

The style you decided to use
```suggestion
    this.authService
      .register(email, password)
      .subscribe(
        () => {
          this.unsuccessWarning = false;
          this.openSuccessSnackBar('Successfully created!');
        },
        () => {
          this.unsuccessWarning = true;
        }
      );
```

As Andrew has said many times, it doesn't matter which one from correct code style approaches you choose, the important thing is to use it consistently throughout your code base.